# params_assign4.yaml
/**:
  ros__parameters:
    mecanum_topic_velocity: "mecanum_velocity"
    mecanum_topic_position: "mecanum_position"
    imu_topic_acceleration: "imu_sim_acceleration"
    imu_topic_velocity: "imu_sim_velocity"
    imu_topic_position: "imu_sim_position"
    rate_hz: 10

sensor_simulator_node:
  ros__parameters:
    intervals:
      0:
        axis: "linear_x"
        poly: "constant"
        t0: 0.0
        t1: 5.0
        y0: 1.0
      1:
        axis: "linear_y"
        poly: "linear"
        t0: 2.0
        t1: 7.0
        y0: 0.0
        y1: 2.0
      2:
        axis: "linear_z"
        poly: "quadratic"
        t0: 1.0
        t1: 4.0
        y0: 0.0
        y1: 1.0
        tm: 2.5
        ym: 2.0
      3:
        axis: "angular_z"
        poly: "linear"
        t0: 0.0
        t1: 6.0
        y0: 0.0
        y1: 0.5

# MECANUM WHEEL PRIMAL DIRECTIONS
#                 FL  FR  RL  RR
# FORWARD     =   1,  1,  1,  1
# RIGHT       =   1, -1, -1,  1
# ROTATE CW   =   1, -1,  1, -1
wheel_velocity_simulator_node:
  ros__parameters:
    intervals:
      0:
        wheel: "wfl"
        poly: "constant"
        t0: 0.0
        t1: 5.0
        y0: 2.0
      1:
        wheel: "wfr"
        poly: "linear"
        t0: 1.0
        t1: 6.0
        y0: 1.0
        y1: 3.0
      2:
        wheel: "wrl"
        poly: "quadratic"
        t0: 2.0
        t1: 7.0
        y0: 0.0
        y1: 1.5
        tm: 4.5
        ym: 2.0
      3:
        wheel: "wrr"
        poly: "linear"
        t0: 0.0
        t1: 4.0
        y0: 0.0
        y1: 2.0
position_approximator_mecanum_node:
  ros__parameters:
    wheel_names: ["wfl", "wfr", "wrl", "wrr"]
    wheel_radius: 0.05  # meters
    robot_length: 0.3   # meters
    robot_width: 0.2    # meters
    start_x: 1.0       # meters
    start_y: 1.0       # meters
    start_yaw: 1.0     # radians
position_velocity_approximator:
  ros__parameters:
    start_x: 0.0       # meters
    start_y: 0.0       # meters
    start_yaw: 0.0     # radians